pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -159,7 +173,30 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 		orchestrationStatusKey, provisioningLabelKey,
 	)
 
-	originalNodes, remainingNodes, completed, statuses, err := c.checkNodesCompleted(selector, config, configOSDs)
+	// For OSDs that are definitely update operations, we want to be able to defer the update until
+	// after we have created new OSDs in order to speed up cluster scale-out. Allow us to defer
+	// updating by keeping a list of deferred configmaps and processing those after creates have
+	// been done. If OSD provisioning doesn't report a "preexisting" status, then no configmaps will
+	// be deferred, and OSD create/update ops will be intermixed in whatever order is processed.
+	deferredConfigMaps := []*v1.ConfigMap{}
```

Comment:
In a future PR how are you envisioning updating the OSDs based on failure domain? Would it be as simple as defining one slice per failure domain, instead of a single slice?

Reply:
I am just trying to make the smallest set of changes to enable this feature right now. All planning for failure domain support is here: https://github.com/rook/rook/pull/6975. 

My thought so far has been to totally separate creation from update. First focus on creating new OSDs. After that, update all OSD deployments. If there is a failure during provisioning that makes a configmap never "completed", then the create step would time out, and code would continue on to the update step.

I am now wondering if it might make sense to have 2 separate goroutines running simultaneously where one creates new OSDs and the other updates existing OSDs by failure domain. But that's for discussion on #6975